docs/notebooks/multifidelity_modelling.pct.py

```diff
@@ -0,0 +1,263 @@
+
+# Create an observer class to deal with multifidelity input query points
+class Observer:
+    def __init__(self):
+
+        self.simulator = simulator
+
+    def __call__(self, x, add_noise=True):
+
+        # Extract raw input and fidelity columns
+        x_input, x_fidelity = check_and_extract_fidelity_query_points(x)
+
+        # note: this assumes that my_simulator broadcasts, i.e. accept matrix inputs.
+        # If not you need to replace this by a for loop over all rows of "input"
+        observations = self.simulator(x_input, x_fidelity, add_noise)
+        return Dataset(query_points=x, observations=observations)
+
+
+# Instantiate the observer
+observer = Observer()
+
+# %% [markdown]
+# Now we can define the other parameters of our problem, such as the input dimension, search space and number of fidelities.
+
+# %%
+from trieste.space import Box
+
+input_dim = 1
+
+lb = np.zeros(input_dim)
+ub = np.ones(input_dim)
+
+n_fidelities = 3
+
+input_search_space = Box(lb, ub)
+
+
+# %% [markdown]
+# ## Create initial dataset
+
+# %%
+# Define sample sizes of low, mid and high fidelities
+sample_sizes = [18, 12, 6]
+
+xs = [tf.linspace(0, 1, sample_sizes[0])[:, None]]
+
+# Take a subsample of each lower fidelity to sample at the next fidelity up
+for fidelity in range(1, n_fidelities):
+    samples = tf.Variable(
+        np.random.choice(xs[fidelity - 1][:, 0], size=sample_sizes[fidelity], replace=False)
+    )[:, None]
+    xs.append(samples)
+
+initial_samples_list = [tf.concat([x, tf.ones_like(x) * i], 1) for i, x in enumerate(xs)]
+initial_sample = tf.concat(initial_samples_list, 0)
+initial_data = observer(initial_sample, add_noise=True)
+
+# %% [markdown]
+# We can plot the initial data. We separate the dataset into individual fidelities using the `split_dataset_by_fidelity` function.
+
+# %%
+from trieste.data import split_dataset_by_fidelity
+
+data = split_dataset_by_fidelity(initial_data, num_fidelities=n_fidelities)
+
+plt.scatter(data[0].query_points, data[0].observations, label="Fidelity 0")
+plt.scatter(data[1].query_points, data[1].observations, label="Fidelity 1")
+plt.scatter(data[2].query_points, data[2].observations, label="Fidelity 2")
+plt.legend()
+plt.show()
+
+# %% [markdown]
+# Now we can fit the `MultifidelityAutoregressive` model to this data. We use the `build_multifidelity_autoregressive_models` to create the sub-models required by the multifidelity model.
+
+# %%
+from trieste.models.gpflow import (
+    MultifidelityAutoregressive,
+    build_multifidelity_autoregressive_models,
+)
+
+# Initialise model
+multifidelity_model = MultifidelityAutoregressive(
+    build_multifidelity_autoregressive_models(initial_data, n_fidelities, input_search_space)
+)
+
+# Update and optimize model
+multifidelity_model.update(initial_data)
+multifidelity_model.optimize(initial_data)
+
+# %% [markdown]
+# ## Plot Results
+#
+# Now we can plot the results to have a look at the fit.
```

Comment:
Here add a sentence about how `predict` works? eg as in training the last column specifies which fidelity you want prediction from